src/vals.c

```diff
@@ -71,22 +61,41 @@ static void vals_compact(struct vals *vals)
     vals->len = j + 1;
 }
 
-static struct vals *vals_from_pairs(struct rill_pairs *pairs)
+static struct vals *from_pairs(struct rill_pairs *pairs, enum rill_col col)
 {
     struct vals *vals =
         calloc(1, sizeof(*vals) + sizeof(vals->data[0]) * pairs->len);
+
     if (!vals) return NULL;
 
     vals->len = pairs->len;
-    for (size_t i = 0; i < pairs->len; ++i)
-        vals->data[i] = pairs->data[i].val;
+    if (col == rill_col_a) {
+        for (size_t i = 0; i < pairs->len; ++i)
+            vals->data[i] = pairs->data[i].key;
+    }
+    else  {
+        for (size_t i = 0; i < pairs->len; ++i)
+            vals->data[i] = pairs->data[i].val;
+    }
 
     vals_compact(vals);
     return vals;
 }
 
+static inline struct vals *vals_from_pairs(struct rill_pairs *pairs)
+{
+    return from_pairs(pairs, rill_col_b);
+}
+
+static inline struct vals *keys_from_pairs(struct rill_pairs *pairs)
+{
+    return from_pairs(pairs, rill_col_a);
+}
+
 static struct vals *vals_merge(struct vals *vals, struct vals *merge)
```

Comment:
I'm pretty sure this function can be deleted.

Reply:
This is still being used in the merge logic